strawberry/annotation.py

```diff
@@ -43,6 +43,9 @@ def __init__(
     ):
         self.annotation = annotation
         self.namespace = namespace
+        if isinstance(annotation, StrawberryAnnotation):
+            self.annotation = annotation.annotation
+            self.namespace = annotation.namespace
```

Comment:
I wonder if we should prevent passing a StrawberryAnnotation to StrawberryAnnotation itself, we could reuse the existing one, what do you think @BryceBeagle ?

Reply:
I'll take a closer look tomorrow, but this looks to be a bandaid solution instead of addressing the underlying problem. I don't think inheritance should lead to a situation where `StrawberryAnnotation` is constructed recursively. I think we should investigate that.

Reply:
Updated with a fix for the root cause; just a typo.

Reply:
This looks good to me just looking at the diff. Good spot. I'll run the changes in a bit to double check this is what we want :slightly_smiling_face: